llvm/lib/MC/RepoObjectWriter.cpp

```diff
@@ -297,11 +302,25 @@ void RepoObjectWriter::recordRelocation(MCAssembler &Asm,
 
     RenamedSymA->setUsedInReloc();
   }
-  Relocations[&FixupSection].emplace_back(FixupOffset, RenamedSymA, Type,
-                                          Addend, SymA, OriginalC);
 
-  if (auto const *Dependent = SymA->CorrespondingTicketNode)
+  const auto *Dependent = SymA->CorrespondingTicketNode;
+  StringRef UsedSymbolName = RenamedSymA->getName();
+  if (Dependent) {
     Dependents[FixupSection.hash()].insert(Dependent);
+    if (Dependent->getLinkage() == GlobalValue::PrivateLinkage) {
+      const std::string SymbolName =
+          Dependent->getPruned()
+              ? Dependent->getNameAsString().str()
+              : MCSymbolRepo::getFullName(Ctx, UsedSymbolName,
+                                          Dependent->getDigest());
+      RenamesTN.insert(std::make_pair(
+          Dependent, llvm::make_unique<std::string>(SymbolName)));
+      UsedSymbolName = StringRef(RenamesTN[Dependent]->c_str());
```

Comment:
> 
> 
> This line performs a search of `RenamesTN` to find the `Dependent` key. `DenseMap::insert()` returns an iterator referencing the inserted position. We should use it.
> 
Fixed it by using the iterator.

> We assign to `UsedSymbolName` twice. Can one be eliminated?
>
I don’t think that one assignment can be eliminated since the `SymbolName` is also used as input parameter (named `InitialName`) of function of `getFullName`.